### ISSUE TITLE
No actualiza mensaje de error

### ISSUE THREAD

**Issue by dark07perez** [Microsoft/vscode#46895]
Issue Type: <b>Bug</b>

Algunas veces, cuando el codigo tiene errores de por ejemplo el constructor mal invocaco o que falta un pareametro, luego de solucionar el error no se desaparece la advertencia de error, y para poder hacer que se quite hay que cerrar el archivo y abrir de nuevo, no es suficiente con guardar o guardar todo.

VS Code version: Code 1.21.1 (79b44aa704ce542d8ca4a3cc44cfca566e7720f1, 2018-03-14T14:46:30.761Z)
OS version: Linux x64 4.13.0-37-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-3630QM CPU @ 2.40GHz (8 x 2394)|
|Load (avg)|3, 2, 1|
|Memory (System)|15.55GB (1.51GB free)|
|Process Argv|/usr/share/code/code --unity-launch|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (11)</summary>

Extension|Author (truncated)|Version
---|---|---
php-debug|fel|1.12.2
php-intellisense|fel|2.3.0
php-pack|fel|1.0.1
teste|ign|2.1.2
mysql-syntax|jak|1.3.1
gradle-language|nac|0.2.3
java|red|0.21.0
vscode-java-debug|vsc|0.7.0
vscode-java-pack|vsc|0.3.0
vscode-java-test|vsc|0.4.0
vscode-maven|vsc|0.6.0

</details>
Reproduces only with extensions

**Comment by dbaeumer** [comment]
@dark07perez can you please provide the description in English.

**Comment by dbaeumer** [comment]
An it is always good to provide steps on how to reproduce the problem you are seeing.

**Comment by justalemon** [comment]
Translation:

```
Sometimes when the code has errors, for example the constructor is invoked incorrectly or with missing parameters, after fixing the error the warning does not disappear, and to make it vanish you should close the file and open it again, is not enough to save or save everything.
```

**Comment by dbaeumer** [comment]
Which language is this happening on. To correctly act on this issue steps to reproduce are necessary. @dark07perez can you please provide these steps.

**Comment by dark07perez** [comment]
Hello, i am sorry, exams time.

I will use an example: 


I have two classes "User" and "user.service", "user.service" has a method "create"

user.ts============================================

export class User { 
	public id_user: number;
	public name: string;

	constructor( id: number, name: string ) {
		this.id = id;
		this.name = name;
	}
}
================================================

user.service.ts=====================================

import { User } from "../models/User";
import { Observable } from "rxjs/Observable";

export class UsersService  {	
        private resourceUrl = 'user/';
	private userCurrent: User;

	constructor(public config: AppConfiguration, public authService: AuthService, public http: Http){                                                           super(config, authService, http); 
this.userCurrent = new User(-1, "" );
   }

	create(number: id, name: string): Observable<User> {	
		return this.post(this.resourceUrl + 'addUser',
			new User(id, name)
		);
	}

}
================================================

Now, i  will talk about error, the situation and solution...

If, for example, i change the constructor of "User" like this: 

user.ts============================================
. . .
	public lastname: string;
constructor( id: number, name: string, lastname: string ) {
		this.id = id;
		this.name = name;
		this.lastname=lastname;
	}
. . .
================================================

So, an error is generated:

" [ts]  Expected 3 arguments, got 4..." 

Now, I'm going to fix the error...

Step 1:

I edited the file again "user.ts" or edited the file "user.service" for to adjust it to the changes

1. I can erase the parameter "lastname" and leave it as it was

2. I can add the "surname" parameter and adjust it to the new constructor


Step 2:

Save all changes.


Issue:

The error message continues to appear in "user.service.ts" " [ts]  Expected 3 arguments, got 4..." 


For this, i close "user.service.ts" and reopen.

Most times this works, but when it does not work i close the program and reopen it. But sometimes the error persists!!!

**Comment by dbaeumer** [comment]
@dark07perez thanks for the steps. Reopening and assigning to @mjbvz

**Comment by dark07perez** [comment]
If you need more information, do not hesitate to let me know

**Comment by mjbvz** [comment]
@dark07perez Can you test this in the latests VSCode insiders as well please. Also, does your project have a `jsconfig.json`?

**Comment by dark07perez** [comment]
Sure, I'll try the latest version, but, I need more time. Usually the problem happens after several tens hours (consecutive) using VSC.

**Comment by dark07perez** [comment]
And, i haven’t a ```jsconfig.json```

**Comment by mjbvz** [comment]
Yes, please try creating a [`jsconfig`](https://code.visualstudio.com/Docs/languages/javascript#_javascript-projects-jsconfigjson) and try this in the latest insiders. If you can share the project I can also investigate this better

**Comment by dark07perez** [comment]
Thanks, i try it.

Project: https://github.com/APPSCO-SOSW/CookAppAngular
